visionppi/app/src/main/java/org/mifos/visionppi/utils/ConnectionDetector.kt

```diff
@@ -0,0 +1,22 @@
+package org.mifos.visionppi.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+
+class ConnectionDetector {
+
+    fun isConnectingToInternet(context: Context): Boolean {
+        val connectivity = context.getSystemService(
+            Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        if (connectivity != null) {
```

Comment:
instead of this, we could just get the `activeNetworkInfo` and check whether it's null or not. If it's null, the device is offline, else it has a network connection. 

For example: 

```kotlin
fun networkAvailable (activity:AppCompatActivity): Boolean {
        val connectivityManager = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return  networkInfo != null && networkInfo.isConnected
}
```